src/queue.js

```diff
@@ -204,4 +204,50 @@ Queue.prototype.shutdown = function() {
   }));
 };
 
+/**
+ * Gets queue worker count
+ */
+Queue.prototype.getWorkerCount = function() {
+  return this.workers.length;
+};
+
+/**
+ * Adds a queue worker
+ */
+Queue.prototype.addWorker = function() {
+  logger.debug('Queue: adding worker');
+  var processId = (this.specId ? this.specId + ':' : '') + this.workers.length;
+  var worker = new QueueWorker(
+    this.tasksRef,
+    processId,
+    this.sanitize,
+    this.suppressStack,
+    this.processingFunction
+  );
+  this.workers.push(worker);
+
+  if (_.isUndefined(this.specId)) {
+    worker.setTaskSpec(DEFAULT_TASK_SPEC);
+  // if the currentTaskSpec is not yet set it will be called once it's fetched
+  } else if (this.currentTaskSpec) {
+    worker.setTaskSpec(this.currentTaskSpec);
+  }
+  return worker;
+};
+
+/**
+ * Removes a queue worker if one exists
+ */
+Queue.prototype.shutdownWorker = function() {
+  logger.debug('Queue: removing worker');
+  var shutdownPromise;
+  var worker = this.workers.shift();
+
+  if (worker) {
+    shutdownPromise = worker.shutdown();
+  }
+  return shutdownPromise;
```

Comment:
Yeap no worries. My intention was to provider a way to tell if a worker was shutdown or not from the call, rather than query totalWorkers before and after. I actually don't need this functionality in my use case so I'm happy to always resolve.

A couple of things I considered:
a. throwing an error if there was no more workers to shutdown.
b. returning a promise that resolved when the shutdown finish with the worker or undefined if there was no more workers to shutdown
c. as you suggested returning a resolved promise

I'm leaning towards a. What do you think @drtriumph